File: 6KYU/Simple Fun #305: Typist/solution.c
unsigned typist(const char *string) {

  int capsOn = 0;
  unsigned clicks = 0;
  
  for(int i = 0; string[i] != '\0'; i++) {
     if(string[i] >= 'A' && string[i] <= 'Z' && capsOn == 0) {
       capsOn = 1;
       clicks += 2;
     } else if(string[i] >= 'a' && string[i] <= 'z' && capsOn == 1) {
       capsOn = 0;
       clicks += 2;
     } else {
       clicks ++;
     }
  }
  
  return clicks;
}

File: 6KYU/Simple Fun #305: Typist/solution.rb
def typist(s)
  caps_on = false
  clicks = 0
  
  for c in s.chars
    if c >= 'A' and c <= 'Z' and !caps_on
      caps_on = true
      clicks += 2
    elsif c >= 'a' and c <= 'z' and caps_on
      caps_on = false
      clicks += 2
    else
      clicks += 1
    end
  end
  
  clicks
end